static/js/components/StarList.jsx

```diff
@@ -21,25 +39,49 @@ const StarList = ({ sourceId }) => {
     fetchStarList();
   }, [sourceId, dispatch]);
 
-  return (
-    <div className={styles.starListDiv}>
-      <code className={styles.starList}>
-        {
-          starList && starList.map((item, idx) => (
-            // eslint-disable-next-line react/no-array-index-key
-            <React.Fragment key={idx}>
-              { item.str }
-              <br />
-            </React.Fragment>
-          ))
+  return starListElem(starList);
+};
+
+export const ObservingRunStarList = ({ observingRunId }) => {
+  const [starList, setStarList] = useState([{ str: 'Loading starlist...' }]);
+  const { assignments } = useSelector((state) => state.observingRun);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const fetchStarList = async () => {
+      const promises = assignments.map(
+        (assignment) => (
+          dispatch(
+            GET(
+              `/api/sources/${assignment.obj_id}/offsets?facility=Keck`, 'skyportal/FETCH_STARLIST'
+            )
+          )
+        )
+      );
+
+      const starlist_info = [];
+      await Promise.allSettled(promises).then(
+        (values) => {
+          values.forEach(
+            (response) => starlist_info.push(...response.value.data.starlist_info)
+          );
         }
-      </code>
-    </div>
-  );
+      );
+      setStarList(starlist_info);
+    };
+    fetchStarList();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [observingRunId, dispatch]);
```

Comment:
`observingRunId` isn't even used in the hook, is it? shouldn't `assignments` be in the deps array?

Reply:
Yeah. I only want to render this once every page load, so I will reimplement this without using `useEffect`.

Reply:
Fixed